Nest start failling when path has spaces

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behavior

Running `nest start` on a folder that the full path has a space throws module not found error. You can use the starter to test that behavior.

```
node:internal/modules/cjs/loader:1147
  throw err;
  ^

Error: Cannot find module '/Users/My'
Require stack:
- internal/preload
    at Module._resolveFilename (node:internal/modules/cjs/loader:1144:15)
    at Module._load (node:internal/modules/cjs/loader:985:27)
    at internalRequire (node:internal/modules/cjs/loader:175:19)
    at Module._preloadModules (node:internal/modules/cjs/loader:1582:5)
    at loadPreloadModules (node:internal/process/pre_execution:731:5)
    at setupUserModules (node:internal/process/pre_execution:172:3)
    at prepareExecution (node:internal/process/pre_execution:132:5)
    at prepareMainThreadExecution (node:internal/process/pre_execution:55:10)
    at node:internal/main/run_main_module:11:19 {
  code: 'MODULE_NOT_FOUND',
  requireStack: [ 'internal/preload' ]
}
```

<img width="2555" alt="image" src="https://github.com/nestjs/nest-cli/assets/2023083/3bdc9c18-bb38-485c-a69f-b8bc6054ebcb">


### Minimum reproduction code

https://github.com/nestjs/typescript-starter

### Steps to reproduce

1. Create a folder that has an space in any location. E.g.: `mkdir /Users/My\ Test`
2. Inside the folder run `nest new test`
3. nest start

### Expected behavior

We should be able to run projects that the full path has spaces.

### Package version

10.2.0

### NestJS version

10.2.7

### Node.js version

20.9.0

### In which operating systems have you tested?

- [X] macOS
- [ ] Windows
- [ ] Linux

### Other

_No response_

--- a/actions/start.action.ts
+++ b/actions/start.action.ts
@@ -177,7 +177,7 @@ export class StartAction extends BuildAction {
     }
     const sourceMapsRegisterPath = this.getSourceMapSupportPkg();
     if (sourceMapsRegisterPath !== undefined) {
-      processArgs.unshift(`-r ${sourceMapsRegisterPath}`);
+      processArgs.unshift(`-r "${sourceMapsRegisterPath}"`);
     }
     return spawn(binaryToRun, processArgs, {
       stdio: 'inherit',

